gum/gummemory.c

```diff
@@ -396,11 +474,55 @@ gum_match_pattern_try_match_on (const GumMatchPattern * self,
         return FALSE;
       }
     }
+    else if (token->type == GUM_MATCH_MASK)
+    {
+      there_are_masks = TRUE;
+    }
+  }
+
+  if (!there_are_masks)
+    return TRUE;
+
+  for (i = 0; i != self->tokens->len; i++)
+  {
+    GumMatchToken * token;
+
+    token = (GumMatchToken *) g_ptr_array_index (self->tokens, i);
+    if (token->type == GUM_MATCH_MASK)
+    {
+      gchar * p;
+
+      p = (gchar *) bytes + token->offset;
+      if (gum_memcmp_mask ((guint8 *) p, (guint8 *) token->bytes->data,
+         (guint8 *) token->masks->data, token->masks->len) != 0)
+      {
+        return FALSE;
+      }
+    }
   }
 
   return TRUE;
 }
 
+static gint
+gum_memcmp_mask (guint8 * haystack,
+                 guint8 * needle,
+                 guint8 * mask,
+                 gsize len)
+{
+  guint i;
```

Comment:
I'd say `i` should be the same type as `len`. Maybe using `guint` for `len` is better since `GArray`'s `len` field is a `guint`.